pennylane/math/quantum.py

```diff
@@ -972,6 +975,101 @@ def sqrt_matrix(density_matrix):
     return vecs @ qml.math.diag(qml.math.sqrt(evs)) @ qml.math.conj(qml.math.transpose(vecs))
 
 
+def sqrt_matrix_sparse(density_matrix):
+    r"""Compute the square root matrix of a sparse density matrix where :math:`\rho = \sqrt{\rho} \times \sqrt{\rho}`

```

Comment:
Is this function focused on density matrices? The block encoding context sounded like we also care about other applications :)

Reply:
Yes. Unfortunately this method can't handle too much beyond positive-definite matrices (not necessary to be Hermitian, even though the usage of `sqrt_matrix` in `BlockEncode` strictly deals with Hermitian cases.)

Reply:
As you will reveal in test suites I added, it is well expected whenever the input matrix has non-positive determinant it fails.

Reply:
Maybe I'm being pedantic here, but could it be useful to state the (Hermitian and) positive definiteness condition instead of saying "density matrix", which has the additional requirement to have unit trace?
The reason I'm suggesting it is that we mention density matrices but then will use it in `BlockEncode` to encode things that are not density matrices :upside_down_face:

Reply:
Yep, I agree with @dwierichs here, and was hinting at this in the previous review too. Declaring intent and support should be clear for where this is to be used.

Reply:
corrected the docstr in latest change

Reply:
This argument is still called `density_matrix`. Can we adapt it as suggested by @dwierichs above to match the Hermitian +ve ?

Reply:
Resolved in another updated thread